DatadogInternal/Sources/NetworkInstrumentation/NetworkInstrumentationFeature.swift

```diff
@@ -78,17 +163,23 @@ extension NetworkInstrumentationFeature {
     ///   - task: The created task.
     ///   - additionalFirstPartyHosts: Extra hosts to consider in the interception.
     func intercept(task: URLSessionTask, additionalFirstPartyHosts: FirstPartyHosts?) {
-        guard let request = task.originalRequest else {
+        guard let originalRequest = task.originalRequest else {
             return
         }
 
-        queue.async {
+        // async update to task leads to race condition where the currentRequest could already be sent to the transport
```

Comment:
I would rephrase to describe what we actually have in the code. Right now it looks like a typo on a first glance.

Something like: `// sync update to task prevents a race condition where the currentRequest could already be sent to the transport`

Reply:
that's fair - updated.